File: algorithms/Codility-master/Codility-master/9_MaximumSliceProblem/MaxSliceSum.py
# 100%
def solution(A):
    # write your code in Python 3.6
    if len(A) == 0: return 0
    if len(A) == 1: return A[0]
    
    count = 0
    max_slice = float("-inf")
    for item in A:
        count = max(item, count + item)
        max_slice = max(max_slice, count)
    return max_slice

File: algorithms/Codility-master/Codility-master/9_MaximumSliceProblem/MaxSliceSum.js
// 100%
function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    if (A.length == 0) return 0;
    if (A.length == 0) return A[0];
    let count = 0;
    let max_slice = -Number.MAX_VALUE;
    for (let i=0; i< A.length; i++){
        count = Math.max(A[i], count + A[i]);
        max_slice = Math.max(max_slice, count);
    }
    return Math.floor(max_slice);
}